cmd/access/access_node_builder.go

```diff
@@ -41,20 +53,94 @@ type AccessNodeBuilder interface {
 	ParticipatesInUnstakedNetwork() bool
 }
 
+type AccessNodeConfig struct {
```

Comment:
I think we should separate this into three categories, instead of putting everything in one huge struct:
* Configs: these are fields that are actually configurable by the user (ie command line flags). They are set once at the beginning of the node initialization (either via flags or later on via constructor), and cannot be changed later
* Modules: These are objects initialized in the `Module` calls. Typically these are dependencies for the components:
* Components: These are `ReadyDoneAware` modules created in the `Component` calls.

At the very least, we should separate out the Configs from everything else, because these should not change during the course of the node initialization.

Reply:
ur 100% right! lemme separate those out.